### ISSUE TITLE
[clang][concepts] Out-of-line defn of constrained member template of class template is incorrectly rejected

### ISSUE THREAD

**Issue by ericniebler** [llvm/llvm-project#62291]
Pretty sure the following should compile (with `-std=c++20`):

```c++
template <class... T>
concept A = true;

template <class F>
struct S {
  template <A<S<F>> T>
  void operator()(T);
};

template <class F>
template <A<S<F>> T>
inline void S<F>::operator()(T) {}
```

Result:

```console
<source>:12:19: error: out-of-line definition of 'operator()' does not match any declaration in 'S<F>'
inline void S<F>::operator()(T) {}
                  ^~~~~~~~
1 error generated.
Compiler returned: 1
```

https://godbolt.org/z/3nnc7rEhb

**Comment by shafik** [comment]
CC @erichkeane @alexander-shaposhnikov this behavior has not changed recently but it looks like it should work.

**Comment by alexander-shaposhnikov** [comment]
this one would be fixed by https://reviews.llvm.org/D146178 , however, D146178 is blocked by a more subtle issue,
it's hard to say when we will be able to land it.

**Comment by cor3ntin** [comment]
Fixed in Clang 17, presumably by D146178
